compiler/ilgen/IlBuilder.cpp

```diff
@@ -770,6 +770,34 @@ IlBuilder::VectorLoadAt(TR::IlType *dt, TR::IlValue *address)
    return returnValue;
    }
 
+/**
+ * @brief IlBuilder::StructFieldAddress IlGen to get the address of a struct field
+ * @param structName is the name of the struct type the field belongs to
+ * @param fieldName is the name of the field itself
+ * @param obj is an IlValue representing the instance (address) of the struct
+ * @return IlValue representing the address of the field in the struct
+ */
+TR::IlValue *
+IlBuilder::StructFieldAddress(const char* structName, const char* fieldName, TR::IlValue* obj) {
+   auto offset = typeDictionary()->OffsetOf(structName, fieldName);
+   auto ptype = typeDictionary()->PointerTo(typeDictionary()->GetFieldType(structName, fieldName));
+   auto addr = Add(obj, ConstInt64(offset));
```

Comment:
While this is "a" correct way to represent the address of a field, I do want to ask about how pervasively you intend to use this api. 

One way to access the field in question is to simply generate an indirect load or store node with the child node being the address node returned by this routine you added using a symbol reference that itself has an offset of zero. 

Another equivalent way to access a struct field is to use a symbol reference having an (potentially non zero) offset corresponding to the struct field on the indirect load or store node and use the "obj" TR::IlValue as the child node. 

The latter is a less verbose representation since we have fewer nodes: no constant offset node and no add node either.